java/java.source.base/src/org/netbeans/modules/java/source/NoJavacHelper.java

```diff
@@ -18,99 +18,34 @@
  */
 package org.netbeans.modules.java.source;
 
-import java.lang.reflect.Field;
-import java.lang.reflect.Method;
-import java.security.ProtectionDomain;
-import java.util.concurrent.atomic.AtomicReference;
-import java.util.logging.Level;
-import java.util.logging.Logger;
 import javax.lang.model.SourceVersion;
-import org.objectweb.asm.ClassWriter;
-import org.objectweb.asm.Opcodes;
-import org.openide.modules.OnStart;
 
 /**
  *
  * @author lahvac
  */
 public class NoJavacHelper {
 
-    private static final AtomicReference<Boolean> hasWorkingJavac = new AtomicReference<>();
-    public static boolean hasWorkingJavac() {
-        Boolean res = hasWorkingJavac.get();
-        if (res != null) {
-            return res;
-        }
+    private static final boolean hasWorkingJavac;
+
+    static {
+        boolean res;
         try {
             SourceVersion.valueOf("RELEASE_17");
             res = true;
         } catch (IllegalArgumentException ex) {
-            //OK
             res = false;
         }
-        hasWorkingJavac.compareAndSet(null, res);
-        return hasWorkingJavac.get();
+        hasWorkingJavac = res;
     }
 
-    public static boolean hasNbJavac() {
-        try {
-            Class.forName("com.sun.tools.javac.comp.Repair");
-            return true;
-        } catch (ClassNotFoundException ex) {
-            //OK
-            return false;
-        }
+    public static boolean hasWorkingJavac() {
+        return hasWorkingJavac;
     }
 
-    @OnStart
-    public static class FixClasses implements Runnable {
```

Comment:
I am a bit on a fence with the removal of the `FixClasses`. The sole purpose of this was to prevent wild linkage exceptions while running on JDK 8 without nb-javac. This is obviously a much smaller issue now that (nb-)javac is a default, but the exceptions are likely to make the IDE hard to use. So I can see an argument for both removing the code and keeping it.

Reply:
roger, lets keep it.